porch/func/internal/podevaluator.go

```diff
@@ -432,7 +432,7 @@ func (pm *podManager) imageDigestAndEntrypoint(image string) (*digestAndEntrypoi
 	if err != nil {
 		return nil, err
 	}
-	img, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(gcrane.Keychain))
```

Comment:
Maybe add the `remote.WithContext(ctx)` as well? 

https://github.com/GoogleContainerTools/kpt/pull/3519/files#diff-0fd72eeec83b1c25062dfba7e2628742b684217250e4789bc48f4bd4f1ecacc3R215
https://github.com/google/go-containerregistry/blob/v0.11.0/pkg/gcrane/options.go#L89

Reply:
Good call - plumbed through the context and added `remote.WithContext`